### PR TITLE
[WIP] feat(API): Create a Signup API v1 - Travellers Support

### PR DESCRIPTION
**Pull request by growebux**
The Traveller / Customer should have a way to create an account on the marketplace platforms. With that being said it's necessary to create an API that supports registration into our systems.

The UX Flow:

1 - User into the signup form
2 - Fill up with personal information
3 - User receives an email for verification/validation - BLOCK - Email platform isn't defined #18 
4 - Once the email is confirmed - BLOCK
5 - Users can log into the marketplace website and set up their wallets. - BLOCK

Exceptions:

While email is not verified user cannot log in with the account created.
ACS:

- [x] Form should collect: first name, last name, password, email
- [x] Email should be unique in our databases.
- [x] We should be able to identify users that have not confirmed their email.

## Action Plan:
-  [x] Create `Tokens` type (Schema)
-  [x] Create input `UserInput` (Schema)
-  [x] Create `signUp(userInput!): Tokens` Mutation (Schema)
-  [x] **Build encryption strategy for passwords(BCRYPT)**
 -  [x] Create a function to `generateHashPassword (password: string) : stringHash` 
 - [x] Create a function to `checkHashPassword(password: string) : boolean`
- [x] **Build Token Strategy (JWT)**
 - [x] Create interface UserInfoToken(Initially just the UserId but we could have more information like email)
 -  [x] Create function`generateToken(userId: mongoId, expiryTime: string): Token`  format expiryTime (`30m`, `20s`, `30d`) 
 -  [x] Create function `decryptToken(Token) : UserInfoToken`
  - [x] Create a controller bound to the `User.singUp(userInput: UserInput): Tokens` model. This function will run all the processes of signUp.

### CODE REVIEW

**Comment by growebux** [comment]
@codedavinci @jeannejx @DongJoSuh 

I talked with Sebas about ticket #18 because it will block this task.

@sebas-led is clarifying the requirements with the Marketing team after it is done, he will update this ticket > #18

File: src/resolvers/index.ts [hunk 1]
```diff
@@ -1,19 +1,27 @@
...
+
   },
 
   Mutation: {
     createPaymentMethod: (_, { paymentMethodName }) => {
       return createPaymentMethod(paymentMethodName)
-    }
+    },
+    // //@ts-ignore
+    // signUp: (_, { userInput }) => {
```
Comment (codedavinci) [inline src/resolvers/index.ts:20]:
```
   signUp: async (_, { userInput }) => {
      return await User.signUp(userInput)
    }
```

**Comment by growebux** [comment]
@codedavinci @jeannejx @DongJoSuh 

I tried to run the codegen to generate the types from entities and auth from the schema file, however, the auth schema didn't get read and the types haven't been generated, only the types from entities got generated. Do you have any tips to solve the problem?

**Comment by growebux** [comment]
@codedavinci I finished the Action Plan tasks.

I did the commits to my PR with each step and test.

**Comment by growebux** [comment]
@codedavinci I having an issue getting access to the signup function on the UserSchema.static. When I ran the query on grapql I get this message:

 "message": "Cannot read properties of undefined (reading 'token')",

This happen  on the file user in the line 163.

**Comment by growebux** [comment]
@codedavinci Could you help me?

**Pull request merged**